Backports for 0.6.1

ararslan opened the pull request:
This is my first time doing this, so it's probably an enormous mess.

vtjnash committed:
better static printing of typemap entries

Ref #21715
(cherry picked from commit 4bcb1bf5205d1fc677a5f62c00ea350dec1991e1)

amitmurthy committed:
Fix reuse of client port on Linux. Implement for OSX.

Ref #21818
(cherry picked from commit 0394f47f455defd5758dc7031266a0eb0adebc57)

Gollor committed:
Fixed relative path include on remote machines

Ref #21832
Squashed for backporting
(cherry picked from commit d86b37cac496e6090061966237272be48a8fc8d2)
(cherry picked from commit 7e7393c1e40ea593c96cc767a46d039ec3a59759)
(cherry picked from commit 6a4acc0423e4c746166fcee0ce8dedf44f36aeaa)
(cherry picked from commit 93cf6d93050465944cde7e6a5cdfc23f8fe28cc6)
(cherry picked from commit 23792943a21e18683ca22e528fda5f33fa780cc9)
(cherry picked from commit 7ff63220cc479f7e36ac338d49adcd5c2415f023)
(cherry picked from commit 49abbf3b24b9658d53f0be9203795f7043b3e475)
(cherry picked from commit eadb699d2088658518f907b83599ffbcb201e03b)
(cherry picked from commit 1ce1d75e59bae7ce99859d47281f62dbb3644e27)

jrevels committed:
redesign type inference outer loop (PR #21677)

removes the global work queue, which allows increasing the precision of cycle detection and resolution, and decreases the need for the threading synchronization lock surrounding inference

updates `inInference` flag usage to be merely a hint for `jl_type_infer` to not bother trying to infer a method (helps avoid accidental infinite recursion over inferring type inference), enable inferring inference

(cherry picked from commit 5847317e5e3cc4d353bbfa2225fcc4b36c0e23cd)

amitmurthy committed:
fix listenany to return correct port number for porthint of 0

Ref #21818
(cherry picked from commit 22720368896f99c7385f3955fcd25c51718feeba)

amitmurthy committed:
added NEWS entry

Ref #21818
(cherry picked from commit fa8c4d2ac0c895de3c3d36924409b44ab6b3f2dd)

vtjnash committed:
Method overwriting by an ambiguity should also invalidate the method cache

fix #21963

Ref #21965
(cherry picked from commit 8bacb65bbab9cbb7da2a266dc384cdd62df372aa)

Keno committed:
Add necessary LLVM patches

Ref #22022
(cherry picked from commit 915910c3e3301891dad754d7443907ca49a9eaa5)

amitmurthy committed:
Return socket correctly in socket_reuse_port (#22017)

* Return socket correctly in socket_reuse_port
* Run reuseport tests only if SO_REUSEPORT is supported.
* Test fix - test for successful addition of workers in distributed test

(cherry picked from commit d3bc3295b4596ed5ed916379cdd8d9cab5e49a02)

bjarthur committed:
fix extrema(A,dim) when length(dim)>1

Ref #22118
(cherry picked from commit 4a470cb174f8083acb79843261b1aab20d2fdf99)

rfourquet committed:
fix base(b, big(0), 0) == "0"

Ref #22133
(cherry picked from commit 994f42c3efee2f5a82fef4cd78ed2cad980f47b4)

vtjnash committed:
prevent Docs.apropos from breaking on random stuff interpolated into docs

Ref #22150
(cherry picked from commit 95677b061cc654e911f2cca1727db238a27ac67a)

JeffBezanson committed:
various improvements to parser (#22161)

- move call to `short-form-function-loc` inside `parse-assignment`
  this is faster and fixes nested cases like `f(x)=g(x)=1`
- fix a case where an `unexpected "foo"` message was missing quotes

- optimize common case of arglists of comma-separated simple tokens
- some other small optimizations

- put code back in precedence order
- separate parse-Nary and parse-comma so they're easier to read
- cleaner code for parse-unary and parse-factor

vtjnash committed:
fix homedir error handling for 1020+ character usernames

or 250 characters for a certain broken OS

Ref #22457
(cherry picked from commit 8c1adc82f8ab17bd26612428984c4c9c135a8550)

fredrikekre committed:
add missing method for eigvecs(A::HermOrSym) (#22453)

(cherry picked from commit 15ee73a4d0b6ee964d6ca75217d2e6ea2470b14f)

fredrikekre committed:
document Upper/LowerTriangular
and clarify that Symmetric/Hermitian are views as well

add references to Upper/LowerTriangular in the manual

Ref #22504
(cherry picked from commit c964958260f6c0dd1baa51bc4bf66ffcfe93c135)

KristofferC committed:
update docs for sysimg to show that default_sysimg_path is a function (#22449)

(cherry picked from commit 3808de924404b44c1c2b8fa00b370e6f9989034f)

stevengj committed:
optimize repeat(string, n) for repeating single ASCII chars (#22462)

(cherry picked from commit b1c2d7357ab1cd4129c5a9049c17be0377276540)

fredrikekre committed:
fix matrix multiplication interaction with HermOrSym and Diagonal (#22474)

(cherry picked from commit 0990d9beaee7cda528defff8bdadf75c578132f2)

kshyatt committed:
I love # Example

Ref #22522
(cherry picked from commit 9c6466afd50c5ffa5ae1fa8ee5e9fd4ecb1749c1)

KristofferC committed:
fix not using type in argument list

Ref #22526
(cherry picked from commit a7c0046c74b4c21411de1cffa3720bec470d3ce3)

KristofferC committed:
add a few doctests to broadcast, parse, intfuncs, number and permuteddimsarray (#22289)

(cherry picked from commit d9d822359a51db06dbc349e08e17cc98ea418246)

KristofferC committed:
test fixed inlining bug #18735 (#22530)

(cherry picked from commit 6218a1f9260a3cc428ae79eee96dd38e77b6ce7b)

KristofferC committed:
test elementwise leftshift on BigInt by vector of ints (#22524)

(cherry picked from commit 65784d73ac1c78410b40b7028584533e4ff3b2fd)

KristofferC committed:
add test for filtering empty iterator (#22528)

(cherry picked from commit 96fca0d796e922827091d638903a3db101b21925)

KristofferC committed:
add some doctest to test (#22287)

* fix spurious newlines in Base.Test printing

* add some doctest to test

(cherry picked from commit 2796b40176dbe489602c5eef9951dfb005316779)

fredrikekre committed:
use where syntax in deprecated.jl (#22527)

(cherry picked from commit ab97a0dd4dcb8732da2d6178d3ac5424698bdb06)

kshyatt committed:
More doctests and cleanup for intfuncs (#22515)

* More doctests and cleanup for intfuncs

* Also more doctests for arrays

* More doctests and Example{s} inserting

* Doctests for DAYS

* Fix some failing doctests

* "fix" failing OffsetArray test

* Address more comments

* fix dropstored!

* Fix leading space

* fix typo

(cherry picked from commit de746144cfe76de27b08874b4e37da2f8d10935f)

JeffBezanson committed:
fix bug in HTML showing of method table of function with parameters

Ref #22586
(cherry picked from commit 2f1d30d3640ed3bd574174cb1159e77176c7c2e4)

mtfishman committed:
Extend definition of StridedReshapedArray (#22429)

* Extend definition of StridedReshapedArray

* Add test

(cherry picked from commit 92ff1bc2e3b6c07e7684a54d8d24c282a8f51fbb)

Commit:
Handle 32 bit architecture when creating float ranges

Ref #22644
(cherry picked from commit b9034e7a92ee0d4991c29fdc8686d9f128b30334)
(cherry picked from commit dd4d8eb6b8aee22b030f19a3012ca096f8903476)
(cherry picked from commit a07f7676f5ac741d58a20d3cf9709ec8da7e9cda)
(cherry picked from commit f502e1c0e3bd03e3542a0bbad0fc52a41e5af08f)
(cherry picked from commit ca60ff5a6420554a5a88317119a171f5923c9b26)
(cherry picked from commit 5018fd6af4fb183751ce33bb1c5c8bdd4937373e)
(cherry picked from commit 8713fa3a51a3ad408509a2d131fbcb0420e72f80)
(cherry picked from commit 7772517ea188440e7e1c5e64ff11908e14629dab)
(cherry picked from commit 4de1a518d08c0cec1a79d118933ef41e42751011)
(cherry picked from commit 7f163ba20ecbd4d7abf190d05927088bd73e28fc)

JeffBezanson committed:
fix #22624, crash in certain circular type definitions

This was caused by expanding a type's field types in the wrong
environment.

Ref #22627
(cherry picked from commit 211e8d156c25a081ebfc1b415358126f54bc83f9)

JeffBezanson committed:
speed up common cases of type application. helps #22593

Ref #22593
(cherry picked from commit 7035cba1e91d56241edb8911ea6ea44051691471)

JeffBezanson committed:
small improvements to (de)serialize perf. helps #22593

also add a `sizehint!` method for `ObjectIdDict`

Ref #22661
(cherry picked from commit 99fd74edce5a7849b9c710c0040a16afa3763bb1)

annimesh2809 committed:
Add efficient hashing of CartesianIndex objects

Ref #22657
(cherry picked from commit a18ffdb92a1bd277695de0f6894e2d2372e7ae67)

fredrikekre committed:
document SparseVector and SparseMatrixCSC (#22541)

rewrite manual chapter to include sparse vectors also
some references to SparseVector/SparseMatrix

(cherry picked from commit ee4c5bcf5864de5aad4e44222cb2990dd8d48dd7)

stevengj committed:
fix graphemes eltype for substrings (fixes #22693)

better grapheme-substring testset name

test substring grapheme eltype

Ref #22694
(cherry picked from commit b13e9d1cca137515a9cfcccbb1c3ec142bb1b6bf)

stevengj committed:
simplified a test

Ref #22694
(cherry picked from commit 5bc6b9664b181a55901b76cdd6006feba13e72f4)

JeffBezanson committed:
fix and simplify subtyping rule for `x <: Type{T}`

Ref #22701
(cherry picked from commit f0e3678d4cfa922341e26ceaf85dba5ce99f0c14)

amitmurthy committed:
fix bug in serialization of nested anonymous functions across processes (#22675)

(cherry picked from commit e90f29db30f81f340d4f36669b27ac5a281e2a7f)

yuyichao committed:
Remove extra JL_GC_POP in emit_a_ccall error path

Ref #22730
(cherry picked from commit 390efc081a8ace8934f57a56f785d8e950d656ea)

JeffBezanson committed:
fix field types of Method type

Ref #22743
(cherry picked from commit 93531aa856e6c25c967ddcf821277680639920a2)

p-zubieta committed:
Fix repeat for arrays of arrays

Ref #22747
(cherry picked from commit e8e475e6651d43c3c46442fceee396e32cfd1ed7)

JeffBezanson committed:
look up constructors by type name in depwarn. fixes #21972

add deprecation_exec.jl for testing deprecations with other `--depwarn` settings

Ref #22764
(cherry picked from commit b1c196e49ab108fad5b41722d3e4f9ff53a8e703)

JeffBezanson committed:
fix `let` bound functions with `where` and return types

Ref #22817
(cherry picked from commit a0f11e5f9adf31e07adbe674db0f699c15f40867)

andreasnoack committed:
Make var(Range) type stable (#22778)

Fixes #22773

(cherry picked from commit e5ca3681ff8d18b11ea7c02c00d8723fb9fb417c)

rfourquet committed:
Base.ndigits0z(0, b) must return 0 (fix #22837)

Ref #22838
(cherry picked from commit 33ffc86c692153cd8c4b974721e66e97ee7984a1)

GunnarFarneback committed:
Bugfix for zero matrix in A_ldiv_B! for QRPivoted (#22831)

* Bugfix for zero matrix in A_ldiv_B! for QRPivoted.

* Address review comments.

(cherry picked from commit bf47fd90f50bfda87fad5a501829e91a60c7d0d0)

JeffBezanson committed:
fix #22908, typemap ordering of `Union` vs. `Type{Union{...}}`.

Ref #22909
(cherry picked from commit 7e37e5d4b319015355e19eb0eadd9cf66c6ede3b)

JeffBezanson committed:
fix `tuple_type_head` and `tuple_type_tail` on `UnionAll`

Ref #22911
(cherry picked from commit 763d008bebbed1231d7ec0aed7d4d2cad40c62a5)

JeffBezanson committed:
fix codegen bug in `===` on two `SimpleVector`s

Ref #22946
(cherry picked from commit 2fca4d3adf2284c8666ef65622c766f5fcf0460d)

vtjnash committed:
mark julia threads as initialized

ensures that cfunction can see that Julia state is initialized on this thread

also upgrades the embedding/examples test to handle exceptions better
(e.g. to actually detect them) while testing for correctness

Ref #22987
(cherry picked from commit 98a2d1d5734597c60f436fe3fd78f09ee735a183)

JeffBezanson committed:
fix backedge restore list order in incremental serializer

Ref #23012
(cherry picked from commit f1535ab1d70b7d2088a446af6a7adea17ed3e6ed)

maleadt committed:
Add a test for #22582/#22022.

Ref #23021
(cherry picked from commit e55b1dbcb9a073d0e35e6c0aa62600437647d404)

vtjnash committed:
fix method lookup intersection error

fix #23024 and improve subtype test macro to always check for this

Ref #23058
(cherry picked from commit fef53134177875fdfa020f46c0aa753faa571cd1)

vtjnash committed:
inference: fix and improve instanceof_tfunc-based queries

fix #23045

Ref #23059
(cherry picked from commit 32e3673d61584c5b7cc0f9434f5836d2467810e2)

stevengj committed:
suppress warning for redefining docstrings if the symbol is in Main (#23019)

suppress warning for redefining docstrings if the symbol is in Main (fixes #23011)

(cherry picked from commit 4d4bdc1831ac0a9eba19b073bc36e7d4ffcb05d1)

musm committed:
Update deprecated tuple type call's in cfunction (#23046)

Update deprecated tuple type call's in cfunction

(cherry picked from commit 73ca642055ef0d54968626843212462025be7931)

fredrikekre committed:
remove mentioning of internal .typ field in Expr objects

Ref #23082
(cherry picked from commit a07d65d35fd00ec629088cfa57650b005ef885a3)

JeffBezanson committed:
fix #23091, function in LHS of destructuring assignment

Ref #23093
(cherry picked from commit 869f35fd3cf4431415d2c899ef1aa0bd99f18e60)

Commit:
Show in doc that thin is a keyword argument. (#23096)

(cherry picked from commit 2700288b5b49bec0a557e367b3b906bda3c30ee7)

rfourquet committed:
symdiff! on IntSet: fix too permissive bounds checking (#23099)

(cherry picked from commit 0b095323ef582c4938f229ce3b0d31e55d91fc9f)

Commit:
Fix typo: subsitution -> substitution. (#23141)

(cherry picked from commit 83007fb8c4c748fb23d38759daafd77fa1d56c2b)

fredrikekre committed:
fix math in condskeel docstring (#23083)

(cherry picked from commit 0f4229d2e744375255fa1b27cb180f678ebf3e76)

amitmurthy committed:
Release a Future only after it has been fetched locally by sending a del_msg. (#23172)

(cherry picked from commit 60bc26e6fcfe134d32a77307a7832584234e7e2b)

omus committed:
Fix style issues with manual METADATA publishing (#23184)

(cherry picked from commit f87dea2d0075596a3548db1451db3594fdcd9422)

fredrikekre committed:
remove .jl extension in Pkg.update (#23214)

* remove .jl extension in Pkg.update

* add test

(cherry picked from commit 4c5cc04156ba074a8baa028c2a8a41b9e70d56ee)

JeffBezanson committed:
fix #23218, lowering error in `(a,) = (b...,)` (#23242)

(cherry picked from commit b5057f0e6593cc158bbe7d15a0f515004e6164bb)

JeffBezanson committed:
fix #23236 (#23243)

comprehension eta-reduction needs to know that dot operators are not
function names.

(cherry picked from commit 37655e3f8fa4a2d2c89bcd51d3afb96feeea528e)

JeffBezanson committed:
fix #23234, parsing `function (x=0) ...` (#23258)

(cherry picked from commit 771bae01cd5a9cce2daaed40d21f8606ef944a75)

timholy committed:
Fix eltype conversion for StepRangeLen{BigFloat} (fixes #23300)

Ref #23348
(cherry picked from commit 1c2e689dd476e507fca2e468c1de82e88cfc0808)

rfourquet committed:
define Symbol(s::Symbol) = s for performance

Before, `Symbol(:symbol)` was calling the generic
`Symbo(x...)` method which in turns calls `string(x...)`;
this is too slow for just returning the argument.

Ref #23381
(cherry picked from commit 3cfd75b849ef60eeeb6b00c325cf40f17278cb77)

KristofferC committed:
no longer warn in code_warntype for unused variables (#23280)

(cherry picked from commit 33fbfad3fd118cb89b63a241f35c92e20d466337)

andreasnoack committed:
Fix Schur Factorization for 0x0 matrices (#23360)

Fixes #23359

(cherry picked from commit db88dd1e906085c14d1cb9f0376c26f0fcf109c8)

KristofferC committed:
add RowVectors to to list of types included in sparse concatinations (#23297)

(cherry picked from commit d00deb7ce962ae76f99e93b4ce97f6a1c4e27f18)

JeffBezanson committed:
fix #23430, counter overflow in subtyping

Ref #23447
(cherry picked from commit 4d4b9ce06de296df3d51646c66192681ca1a8788)

JeffBezanson committed:
fix bug in printing generator expressions with n-d ranges

Ref #23480
(cherry picked from commit f9527b7ee0cb07384312f32871cbcb960ed72c93)

vtjnash committed:
test-codegen: make jl_dump_compiles test more reliable (#23486)

This was a flaky test, and it's not entirely clear what it was testing.

Common failure was:
        FAILURE
    Error in testset codegen:
    Test Failed
      Expression: tempty == true
       Evaluated: false == true
    ERROR: LoadError: Test run finished with errors
    while loading C:\projects\julia\julia-a661736be5\share\julia\test\runtests.jl, in expression starting on line 29
    Command exited with code 1

(cherry picked from commit 46bff1bf83f82107369b23c3163a53aed4f9801c)

fredrikekre committed:
fix documentation for IOBuffer, close #23398 (#23514)

(cherry picked from commit b729e58dff9c9cfa2640b8c4dcc71dbc91b29871)

JeffBezanson committed:
fix #23558, bug in recursive let-bound functions

Ref #23561
(cherry picked from commit aee1aab70f8521dcebb76ef91173c8c416a8611c)

omus committed:
Display STDOUT from Pkg.build

Ref #23601
(cherry picked from commit 1ca3510baa14e0371fb53764276090f9108b1e9a)

ararslan commented:
I need to redo this from scratch. Things to do differently:
* Cherry pick individual commits; don't use merge commits unless it was a squash merge.
* Do not include commits from PRs that still have a [needs X] label for any X.

ararslan commented:
Looks like something was backported incorrectly, likely a bad conflict resolution. Not sure what exactly is causing it but it looks like it's failing somewhere around src/ccall.cpp. @vtjnash, would you be willing to take a look at that and make sure that part looks okay?

vtjnash commented:
Added #22161, dropped PR #22472 (096e13d2ad938ecf87bc95aa02557ac3982f4a18 ("Store global variables and function addresses as 32bit offsets") and 054618e ("Fix function base address lookup on Win32 and FreeBSD")),
rebased on top of #23683, and corrected minor typos. Still getting some failures in linalg/generic, linalg/cholesky, misc, linalg/qr, and linalg/diagonal, and misc which you'll need to resolve.

ararslan commented:
Ack, I merged #23683 and now there are conflicts. 😑 The linalg stuff may be due to #22138, which Fredrik has since advised against backporting.

ararslan commented:
In backporting #22945, it appears the change actually does not fix the type instabilities observed on this branch. @andreasnoack, do you know whether there are any other accompanying changes that should be backported in order to make `vecnorm` inferrable on 0.6.1? Or should I drop that commit and we can revisit it for 0.6.2?

iamed2 committed:
Make print_matrix O(1) again (#23681)

(cherry picked from commit c23d6bc0db0bddcc9d83a1d3609975a5dcc8fff8)

ararslan committed:
Adjust capitalization in LibGit2 test to be version agnostic

mbauman committed:
Fix missing bounds checks for trailing zero dimensions (#23723)

This fixes issue #23629 for 0.6. It is done independently from the fix for master (in #23628) due to all the deprecation changes.

ararslan committed:
Move write_build to the outer scope

bmharsha commented:
Are you planning to share links to pre-release binaries of Julia v0.6.1?

ararslan commented:
I'm not sure. Probably not. Unless that's something we usually do?

StefanKarpinski commented:
@bmharsha, do you need them for preparing JuliaPro?

ararslan commented:
@nanosoldier `runbenchmarks(ALL, vs=":release-0.6")`

ararslan commented:
Anyone know how to run PkgEval here?

ViralBShah commented:
Would be great to trigger a `PkgEval` build from here like we do for benchmarks.

jrevels commented:
> Would be great to trigger a PkgEval build from here like we do for benchmarks.

ref https://github.com/JuliaCI/Nanosoldier.jl/issues/1 

Nanosoldier's job submission system is actually generic - it just happens to only have one job type implemented at the moment (running benchmarks).

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/23670#issuecomment-330964549) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/285e50ce47e1a4db6ca373ef6095704efdc769c4/737a987_vs_4aa661a/report.md). cc @ararslan

maleadt committed:
Add a small patch that allows us to generate code for PTX ISA 6.0.

This is required for supporting newer NVIDIA GPUs, running under CUDA 9.0.

Ref #23817
(cherry picked from commit 6626c1a153aba792dbc57bac614a64b1af692e71)

ararslan commented:
Sorry for the delay here, folks. There are some things I'll need some help with and/or guidance on to be able to move forward on this:

* Build binaries from this branch (I guess from build.julialang.org?)
* Run PackageEvaluator to compare release-0.6 to this branch (is that on a Nanosoldier non-benchmarking node?)
* What to do once everything comes back clean and this is merged into release-0.6

ViralBShah commented:
@staticfloat may be able to help on a couple of those things.

staticfloat commented:
I will help once I'm back from brazil at the beginning of October.  Alex, if you don't have login access to build.julialang.org message me and I'll get you a login.

ararslan commented:
I do have access, I just don't know what to do with it 😄

staticfloat commented:
When you have a git sha you want to build, plug it into the package
builders. You can only do that if you're logged in.

On Wed, Sep 27, 2017, 15:10 Alex Arslan <notifications@github.com> wrote:

> I do have access, I just don't know what to do with it 😄
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaLang/julia/pull/23670#issuecomment-332608620>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAH_aEOqPHFkx6thbso5PRS327yIwsUjks5smo-SgaJpZM4PT6CX>
> .
>
-- 

-E

ararslan commented:
Wouldn't the Git SHA I want to build be the SHA of the merge commit of this branch into release-0.6? But we shouldn't merge this until PkgEval has been run, so I'm unclear on which SHA should be used.

yuyichao commented:
I'm 99% sure branch SHA works and I've used that many times.

ararslan commented:
Would I use "Build and test" under "Force All Builds," supplying `aa/backports-0.6` as the branch?

KristofferC commented:
Would be good to also run the doctests if that hasn't been done.

fredrikekre committed:
fix doctests for 0.6.1 (#23916)

ararslan commented:
Figuring out PackageEvaluator was a bit of a challenge but it seems that the packages that failed on 0.6.1 but passed on 0.6.0 failed for unrelated reasons. So it looks like 0.6.1 is nonbreaking. :)

ararslan commented:
Once more for good measure (and to figure out whether any of the regressions are real): @nanosoldier `runbenchmarks(ALL, vs=":release-0.6")`

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/23670#issuecomment-334662541) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/93fd1d0a5893e92c14a80e4e51fc225970ec7955/21b9b6f_vs_4aa661a/report.md). cc @ararslan

fredrikekre commented:
`repeat` regression is real:
v0.6.0:
```julia
julia> @btime repeat($A, inner = $[24], outer=$[1]);
  51.796 μs (611 allocations: 56.66 KiB)
```
This branch:
```julia
julia> @btime repeat($A, inner=$[24], outer=$[1]);
  290.690 μs (3811 allocations: 150.41 KiB)
```

fredrikekre commented on base/abstractarraymath.jl:
```diff
@@ -415,7 +415,7 @@ _reperr(s, n, N) = throw(ArgumentError("number of " * s * " repetitions " *
                 inner_indices[i] = (1:n) + ((c[i] - 1) * n)
             end
-            R[inner_indices...] = A[c]
+            fill!(view(R, inner_indices...), A[c])
```
This must be the cause of the repeat regression, @pabloferz

pabloferz replied:
This is a necessary fix, it's meant for the case when `A[c]` is itself an array. I can think of a way of trying to preserve the speed for the cases when `A[c]` is not an array, but I don't know what would be the action to take with regards to the backport.

pabloferz replied:
https://github.com/JuliaLang/julia/pull/24022

p-zubieta committed:
Restore repeat preformance for arrays of scalars (#24022)

(cherry picked from commit af5abb3516d5fb9262a33aafe25cc6f1ff1513bc)

ararslan commented:
Pablo's PR bettered the performance on master, so let's see what effect it had here.

@nanosoldier `runbenchmarks("misc", vs=":release-0.6")`

mbauman commented:
Could I kindly request one more patch to make this round?  #23681 would be nice to backport.  I don't think I see it listed and it's a very trivial fix.

ararslan commented:
Can do 👍

ararslan commented:
Will merge once CI is green unless someone beats me to it. DO NOT SQUASH!

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/23670#issuecomment-334878279) has completed - no performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/1eeea9f0faaa7213a6ecdab342b543a0c3be918d/53fbfff_vs_4aa661a/report.md). cc @ararslan

mauro3 commented:
Thanks! Now we just need the tag v0.6.1.

tkelman commented:
Where are your pkgeval results and logs? Coverage.jl fails tests on this branch and it's definitely related, the parser changes were breaking and shouldn't be backported.

StefanKarpinski commented:
Are you talking about these parser changes: https://github.com/JuliaLang/julia/pull/22161. They were non-breaking according to @JeffBezanson (who seems like a fairly reliable character), and tagged for backporting by @vtjnash. It's possible they were actually breaking, but if so it was subtle. There was a PkgEval run done, and I'm sure @ararslan can post the results here.

ViralBShah commented:
It would be great to hook up PkgEval the way the benchmarks are hooked up - so it can be triggered from here.

StefanKarpinski commented:
Automating the running of PkgEval is an excellent idea, similar to nanosoldier. I have to confess I have no idea where the PkgEval code even lives. PkgEval.jl doesn’t exist and there seems to be zero documentation of how to run it.

ararslan commented:
https://github.com/JuliaCI/PackageEvaluator.jl

PkgEval is just what the cool kids are calling it. 😉

ararslan commented:
> Where are your pkgeval results and logs?

On Nanosoldier, where they ran. I didn't realize I was supposed to post them somewhere, but I can do that.

ararslan commented:
PkgEval for 0.6.1: https://gist.github.com/ararslan/98ef9116c7bb6b5c7dcb041c9f92a2ef

tkelman commented:
Doesn't make much sense that those logs seem to say Coverage.jl passed on this branch. It definitely fails locally against latest release-0.6. That PR caused a change in coverage measurement that made Coverage start failing on nightly when it was merged. Something may be flawed about how the backport test binary that pkgeval used was built.

StefanKarpinski commented:
> PkgEval is just what the cool kids are calling it.

Given that everyone refers to it as `PkgEval` perhaps we should rename the package?

ViralBShah commented:
To make sure I understand correctly, is it that the parser changes seem to have broken Coverage.jl?

ararslan commented:
It appears that this is what changed in Coverage: https://github.com/JuliaCI/Coverage.jl/pull/136. It seems that change resulted in something more correct for Coverage but because of that broke a test that expected the old behavior. The change seems subtle enough that I doubt pretty much any other package would have noticed. It's still odd that Coverage on PkgEval didn't see it though.

ararslan commented:
Interesting, downloading the binaries I used for running PkgEval and doing `Pkg.test("Coverage")` manually fails.
